Merge 3.3.x with protx changes to master

zebra-lucky opened the pull request:
git cherry-pick `
changes from upstream electrum3.2.4 to 3.3.4` to master (manual conflicts resolving)
then merge `upstream-changes` branch to master (manual conflicts resolving)

jardev commented on electrum_dash/gui/qt/network_dialog.py:
```diff
@@ -509,10 +525,14 @@ def __init__(self):
-            if TorDetector.is_tor_port(p):
-                self.found_proxy.emit(("127.0.0.1", p))
-                return
+        while True:
```
Infinite loop after tor detection

jardev commented on electrum_dash/gui/qt/protx_wizards.py:
```diff
@@ -1450,12 +1450,12 @@ def __init__(self, parent, mn=None, start_id=None):
         self.setPixmap(QWizard.LogoPixmap, logo)
         self.setWindowTitle(title)
-        self.setWindowIcon(QIcon(':icons/electrum-dash.png'))
+        self.setWindowIcon(QIcon('electrum-dash.png'))
```
shouldn't be `icon_path(QIcon('electrum-dash.png'))` or read_QIcon('electrum-dash...?

jardev commented on electrum_dash/plugins/revealer/qt.py:
```diff
@@ -48,60 +51,87 @@ def __init__(self, parent, config, name):
         return EnterButton(_('Printer Calibration'), partial(self.calibration_dialog, window))
 
+    def password_dialog(self, msg=None, parent=None):
+        from electrum.gui.qt.password_dialog import PasswordDialog
```
incorrect import - see `electrum`, should be renamed

jardev commented on electrum_dash/util.py:
```diff
@@ -137,11 +152,19 @@ def __new__(cls, value):
 
     def __repr__(self):
-        return 'Duffs(%d)'%self.value
+        return 'Satoshis(%d)'%self.value
```
Satoshis should be renamed to Duffs

jardev commented:
Also look at [wallet.py:1191](https://github.com/akhavr/electrum-dash/blob/95b52393a85cf66f11d0f76e189a12e41c79efcb/electrum_dash/wallet.py#L1191) - '*' was removed from imports, so `public_key_to_p2pkh` should be fixed

jardev commented:
[masternode_manager.py:221](https://github.com/akhavr/electrum-dash/blob/95b52393a85cf66f11d0f76e189a12e41c79efcb/electrum_dash/masternode_manager.py#L221) - `conf` should be replaced with `tx_height.conf`

jardev commented:
[masternode_manager.py:412](https://github.com/akhavr/electrum-dash/blob/95b52393a85cf66f11d0f76e189a12e41c79efcb/electrum_dash/masternode_manager.py#L412) - `proposal_hash` is not initialized

jardev commented:
[protx.py:587](https://github.com/akhavr/electrum-dash/blob/95b52393a85cf66f11d0f76e189a12e41c79efcb/electrum_dash/protx.py#L587) - are you sure it shouldn't be `self.network.interface.server` instead?

zebra-lucky commented:
> Also look at wallet.py:1191 - '*' was removed from imports, so public_key_to_p2pkh should be fixed

> masternode_manager.py:221 - conf should be replaced with tx_height.conf

> masternode_manager.py:412 - proposal_hash is not initialized

> incorrect import - see electrum, should be renamed

> Satoshis should be renamed to Duffs

fixed

> protx.py:587 - are you sure it shouldn't be self.network.interface.server instead?

removed call to `connection_down` at all

Current code or `run` method in `TorDetector` class not looks like need to be fixed:

```Python
    def run(self):
        # Probable ports for Tor to listen at
        ports = [9050, 9150]
        while True:
            for p in ports:
                if TorDetector.is_tor_port(p):
                    self.found_proxy.emit(("127.0.0.1", p))
                    break
            else:
                self.found_proxy.emit(None)
            time.sleep(10)
```

jardev commented:
OK, looks like everything has been fixed

jardev commented:
Looks good
